### PR TITLE
Fixing the cluster spec output formatting issue

### PR DESCRIPTION
**Pull request by viggy28**
To address #723.

### CODE REVIEW

**Comment by viggy28** [comment]
Hi @sgotti, thanks for looking. Updated it.

**Comment by sgotti** [comment]
@viggy28 Thinking and testing this PR I noticed that it will break the `stolonctl spec --defaults` output hiding some defaults values.

One possible solution will require the creation (in stolonctl_spec.go), of two different cluster spec structs (identical to the original cluster spec, one with all json tags with omitempty and one without omitempty) that could be [converted](https://golang.org/ref/spec#Conversions) from the original clusterspec so their marshalling will work for both cases (this will also help detecting when we forgot to add new fields to all the structs at compilation time).

**Comment by viggy28** [comment]
Thanks @sgotti. Updated the PR. Let me know if that works.

File: internal/cluster/cluster.go [hunk 1]
```diff
@@ -290,6 +290,86 @@ type ClusterSpec struct {
 	AutomaticPgRestart *bool `json:"automaticPgRestart"`
 }
 
+type ClusterSpecNew struct {
```
Comment (sgotti) [inline internal/cluster/cluster.go]: There's the need of two different cluster spec structs identical to the original cluster spec:

* one with all json tags with omitempty for normal output
* one with all json tags without omitempty for --default output

File: internal/cluster/cluster.go [hunk 2]
```diff
@@ -290,6 +290,86 @@ type ClusterSpec struct {
...
+	// Interval to wait before next check
```
Comment (sgotti) [inline internal/cluster/cluster.go]: Just remove all the fields comments.

Comment (viggy28) [reply internal/cluster/cluster.go]: updated.

**Comment by sgotti** [comment]
> One possible solution will require the creation (in stolonctl_spec.go), of two different cluster spec structs (identical to the original cluster spec, one with all json tags with omitempty and one without omitempty) that could be converted from the original clusterspec so their marshalling will work for both cases (this will also help detecting when we forgot to add new fields to all the structs at compilation time).

@viggy28 Sorry if I wasn't clear but I meant to use two new structs inside stolonctl/cmd/spec.go, take a look at this compare:

https://github.com/sorintlab/stolon/compare/master...sgotti:stolonctl_spec_improve_output?expand=1

**Comment by sgotti** [comment]
@viggy28 Thanks for your PR, I created a PR #736 with an updated implementation.

**Pull request closed without merging**